### ISSUE TITLE
Question about self.accelerator.backward(loss)

### ISSUE THREAD

**Issue by Chen-Suyi** [lucidrains/denoising-diffusion-pytorch#251]
I think what you want to do is to accumulate the gradient here. However, the loss back-propagated seems not the one accumulated.
![image](https://github.com/lucidrains/denoising-diffusion-pytorch/assets/141543096/c66f34a8-e779-42a6-9568-8c1dd8f53559)

I mean, is there anything wrong with the [line 1016]( https://github.com/lucidrains/denoising-diffusion-pytorch/blob/3f2ef250dcc57e2f1ec29a7c6a98430c0d8da5ef/denoising_diffusion_pytorch/denoising_diffusion_pytorch.py#L1016)?

Using `self.accelerator.backward(total_loss)` seems better.

**Comment by lhaippp** [comment]
oh in this way, it seems like "gradient_accumulate_every " not work

**Comment by lucidrains** [comment]
yea, total loss is just for logging

**Comment by Chen-Suyi** [comment]
> yea, total loss is just for logging

I got your point. loss.backward() will automatically accumulate the gradients if there is no opt.zero_grad(). I misunderstood it at that time.
